commit 5de956c6c6d91a82ca8b13ed4625ad30f0399a6d (refs/changes/78/2978/1)
Author: Marsell Kukuljevic <marsell@joyent.com>
Date:   2017-11-22T00:25:27+13:00 (1 year, 11 months ago)
    
    PUBAPI-1462: Tests not terminating when run without SDC_SETUP_TESTS

diff --git a/.gitignore b/.gitignore
index 3d47f02..8cd521a 100644
--- a/.gitignore
+++ b/.gitignore
@@ -1,3 +1,4 @@
+*.swp
 build
 *.node
 *.pdf
diff --git a/lib/app.js b/lib/app.js
index 2d150fd..11802d1 100644
--- a/lib/app.js
+++ b/lib/app.js
@@ -732,6 +732,18 @@ module.exports = {
                         clients.mahi.close();
                     }
 
+                    if (clients.cns) {
+                        clients.cns.close();
+                    }
+
+                    if (clients.volapi) {
+                        clients.volapi.close();
+                    }
+
+                    process._getActiveHandles().forEach(function (h) {
+                        return h.unref && h.unref();
+                    });
+
                     clients.ufds.close(function (err) {
                         if (err) {
                             throw err;
